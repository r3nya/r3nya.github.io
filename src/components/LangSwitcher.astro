---
import type { Locale } from '../i18n/config';
import { locales } from '../i18n/config';
const { current }: { current: Locale } = Astro.props;
---
<nav class="fixed bottom-4 left-1/2 -translate-x-1/2 transform z-50 text-sm text-palette-3-light dark:text-palette-3-dark bg-palette-4-light/70 dark:bg-palette-4-dark/70 backdrop-blur px-4 py-2 rounded-full shadow">
  <ul class="flex gap-4">
    {locales.map((code) => (
      <li>
        {code === current ? (
          <span class="font-semibold uppercase">{code}</span>
        ) : (
          <a class="uppercase hover:underline" href={`/${code}/`}>{code}</a>
        )}
      </li>
    ))}
  </ul>
</nav>
