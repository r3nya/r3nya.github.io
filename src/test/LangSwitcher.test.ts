// @vitest-environment happy-dom
import { describe, it, expect } from 'vitest';
import LangSwitcher from '../components/LangSwitcher.astro';
import { renderAstroComponent } from './helpers';

describe('LangSwitcher', () => {
  it('renders with English as current language', async () => {
    const result = await renderAstroComponent(LangSwitcher, {
      props: {
        current: 'en',
      },
    });

    expect(result.innerHTML).toMatchSnapshot();
  });

  it('renders with Russian as current language', async () => {
    const result = await renderAstroComponent(LangSwitcher, {
      props: {
        current: 'ru',
      },
    });

    expect(result.innerHTML).toMatchSnapshot();
  });

  it('renders with Spanish as current language', async () => {
    const result = await renderAstroComponent(LangSwitcher, {
      props: {
        current: 'es',
      },
    });

    expect(result.innerHTML).toMatchSnapshot();
  });
});
