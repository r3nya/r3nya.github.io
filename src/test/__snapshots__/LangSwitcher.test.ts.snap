// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LangSwitcher > renders with English as current language 1`] = `"<nav aria-label="Language switcher" class="fixed bottom-4 left-1/2 -translate-x-1/2 transform z-50 text-sm text-palette-3-light dark:text-palette-3-dark bg-palette-4-light/70 dark:bg-palette-4-dark/70 backdrop-blur px-4 py-2 rounded-full shadow" data-testid="lang-switcher"> <ul role="list" class="flex gap-4"> <li role="listitem"> <span class="font-semibold uppercase" aria-current="true" data-lang="en"> en </span> </li><li role="listitem"> <a class="uppercase hover:underline" href="/ru/" lang="ru" hreflang="ru" data-lang="ru" aria-label="Switch language to RU"> ru </a> </li><li role="listitem"> <a class="uppercase hover:underline" href="/es/" lang="es" hreflang="es" data-lang="es" aria-label="Switch language to ES"> es </a> </li> </ul> </nav>"`;

exports[`LangSwitcher > renders with Russian as current language 1`] = `"<nav aria-label="Language switcher" class="fixed bottom-4 left-1/2 -translate-x-1/2 transform z-50 text-sm text-palette-3-light dark:text-palette-3-dark bg-palette-4-light/70 dark:bg-palette-4-dark/70 backdrop-blur px-4 py-2 rounded-full shadow" data-testid="lang-switcher"> <ul role="list" class="flex gap-4"> <li role="listitem"> <a class="uppercase hover:underline" href="/en/" lang="en" hreflang="en" data-lang="en" aria-label="Switch language to EN"> en </a> </li><li role="listitem"> <span class="font-semibold uppercase" aria-current="true" data-lang="ru"> ru </span> </li><li role="listitem"> <a class="uppercase hover:underline" href="/es/" lang="es" hreflang="es" data-lang="es" aria-label="Switch language to ES"> es </a> </li> </ul> </nav>"`;

exports[`LangSwitcher > renders with Spanish as current language 1`] = `"<nav aria-label="Language switcher" class="fixed bottom-4 left-1/2 -translate-x-1/2 transform z-50 text-sm text-palette-3-light dark:text-palette-3-dark bg-palette-4-light/70 dark:bg-palette-4-dark/70 backdrop-blur px-4 py-2 rounded-full shadow" data-testid="lang-switcher"> <ul role="list" class="flex gap-4"> <li role="listitem"> <a class="uppercase hover:underline" href="/en/" lang="en" hreflang="en" data-lang="en" aria-label="Switch language to EN"> en </a> </li><li role="listitem"> <a class="uppercase hover:underline" href="/ru/" lang="ru" hreflang="ru" data-lang="ru" aria-label="Switch language to RU"> ru </a> </li><li role="listitem"> <span class="font-semibold uppercase" aria-current="true" data-lang="es"> es </span> </li> </ul> </nav>"`;
